src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -53,20 +53,48 @@ private static bool IsEnabled()
                    s_diagnosticListener.IsEnabled();
         }
 
-        private static Activity? CreateActivity(HttpRequestMessage requestMessage)
+        private static Activity? StartActivity(HttpRequestMessage request)
         {
             Activity? activity = null;
             if (s_activitySource.HasListeners())
             {
-                activity = s_activitySource.CreateActivity(DiagnosticsHandlerLoggingStrings.ActivityName, ActivityKind.Client);
+                Uri? requestUri = request.RequestUri;
+                if (requestUri?.IsAbsoluteUri == false)
+                {
+                    requestUri = null;
```

Comment:
is there a valid use-case when it could happen? (should we track it in otel http semconv)?

Reply:
This can happen if we use `HttpClientHandler` or `SocketsHttpHandler` with `HttpMessageInvoker`, but such calls then will fail in the underlying platform handler, so I don't see how it can be a valid use case.

Regardless, the existing `DiagnosticsHandler` logic was prepared for relative Uri-s for whatever reason. I did not want to touch that. Maybe we should, will ask folks.